test/unit-test/core_sntp_client_utest.c

```diff
@@ -0,0 +1,439 @@
+
+    /* Pass valid transport interface object but invalid members. */
+    transportIntf.recvFrom = NULL;
+    TEST_ASSERT_EQUAL( SntpErrorBadParameter,
+                       Sntp_Init( &context,
+                                  testServers,
+                                  sizeof( testServers ) / sizeof( SntpServerInfo_t ),
+                                  testBuffer,
+                                  sizeof( testBuffer ),
+                                  dnsResolve,
+                                  getTime,
+                                  setTime,
+                                  &transportIntf,
+                                  NULL ) );
+    transportIntf.recvFrom = UdpRecvFrom;
+    transportIntf.sendTo = NULL;
+    TEST_ASSERT_EQUAL( SntpErrorBadParameter,
+                       Sntp_Init( &context,
+                                  testServers,
+                                  sizeof( testServers ) / sizeof( SntpServerInfo_t ),
+                                  testBuffer,
+                                  sizeof( testBuffer ),
+                                  dnsResolve,
+                                  getTime,
+                                  setTime,
+                                  &transportIntf,
+                                  NULL ) );
+
+    /* Set the transport interface object to be valid for next test. */
+    transportIntf.sendTo = UdpSendTo;
+
+    /* Pass valid authentication interface object but invalid members. */
+    authIntf.generateClientAuth = NULL;
+    TEST_ASSERT_EQUAL( SntpErrorBadParameter,
+                       Sntp_Init( &context,
+                                  testServers,
+                                  sizeof( testServers ) / sizeof( SntpServerInfo_t ),
+                                  testBuffer,
+                                  sizeof( testBuffer ),
+                                  dnsResolve,
+                                  getTime,
+                                  setTime,
+                                  &transportIntf,
+                                  &authIntf ) );
+    authIntf.generateClientAuth = generateClientAuth;
+    authIntf.validateServer = NULL;
+    TEST_ASSERT_EQUAL( SntpErrorBadParameter,
+                       Sntp_Init( &context,
+                                  testServers,
+                                  sizeof( testServers ) / sizeof( SntpServerInfo_t ),
+                                  testBuffer,
+                                  sizeof( testBuffer ),
+                                  dnsResolve,
+                                  getTime,
+                                  setTime,
+                                  &transportIntf,
+                                  &authIntf ) );
+}
+
+
+/**
+ * @brief Test @ref Sntp_Init API correctly initializes a context.
+ */
+void test_Init_Nominal( void )
+{
+#define TEST_SNTP_INIT_SUCCESS( pAuthIntf )                                                               \
```

Comment:
Is it necessary for this to be a macro? They are neat but you don't get any type safety with it.

Reply:
If an incorrect parameter type is passed as the macro parameter, the implementation of the macro will generate a warning as the following which will do the same work as the type safety provided by a function
```
:
/home/ubuntu/Repos/coreSNTP/test/unit-test/core_sntp_client_utest.c: In function ‘test_Init_Nominal’:
/home/ubuntu/Repos/coreSNTP/test/unit-test/core_sntp_client_utest.c:438:29: warning: passing argument 10 of ‘Sntp_Init’ from incompatible pointer type [-Wincompatible-pointer-types]
     TEST_SNTP_INIT_SUCCESS( &testServers );
                             ^
```